provider/pkg/provider/provider.go

```diff
@@ -873,7 +873,8 @@ func (p *googleCloudProvider) Delete(_ context.Context, req *rpc.DeleteRequest)
 		return &empty.Empty{}, nil
 	}
 
-	resp, err := p.client.RequestWithTimeout(res.Delete.Verb, uri, nil, 0)
+	resp, err := retryRequest(p.client, res.Delete.Verb, uri, nil)
```

Comment:
👍🏾  - I guess we need to add this to other crud methods as well?

Reply:
Yep, exactly. I'm testing locally with Delete at the moment, and then I'll roll it out to the other operations when I've verified it's working as expected.

Reply:
Thanks for pursuing this approach. Would also be good to update the GKE test with something that adds a nodepool or something in second step which should trigger the repairing state as an integration test for this.

Reply:
> Would also be good to update the GKE test with something that adds a nodepool or something in second step which should trigger the repairing state as an integration test for this.

We chatted offline about this and decided to defer this until after the planned refactoring. I tested this workflow locally, and it takes ~30 minutes to run, so we need to be mindful of how to include this in CI.